Mobile/ShopLens/Droid/Activities/MainActivity.cs

```diff
@@ -103,59 +115,145 @@ protected override void OnCreate(Bundle savedInstanceState)
                         break;
                     case Resource.Id.NavItemShoppingCart:
                         var intentCart = new Intent(this, typeof(ShoppingCartActivity));
+                        intentCart.PutExtra(noTalkBackIntentKey, noTalkBack);
                         StartActivity(intentCart);
                         break;
                     case Resource.Id.NavItemShoppingList:
                         var intentList = new Intent(this, typeof(ShoppingListActivity));
+                        intentList.PutExtra(noTalkBackIntentKey, noTalkBack);
                         StartActivity(intentList);
                         break;
-                    case Resource.Id.NavItemTextToSpeech:
-                        var intentTTS = new Intent(this, typeof(TextVoicerActivity));
-                        StartActivity(intentTTS);
-                        break;
-                    case Resource.Id.NavItemSpeechRecogniser:
-                        var intentSpeech = new Intent(this, typeof(SpeechActivity));
-                        StartActivity(intentSpeech);
-                        break;
                 }
             };
         }
 
-        void RecogniseVoice(object sender, EventArgs e)
+        protected override void OnStart()
+        {
+            base.OnStart();
+            tutorialNeeded = IsTutorialNeeded();
+        }
+
+        protected override void OnStop()
         {
-            if (commandRecognizer.IsValueCreated)
+            // Get rid of TTS.
+            if (tts != null)
             {
-                speechIntent = VoiceRecognizerHelper.SetUpVoiceRecognizerIntent();
-                commandRecognizer.Value.StartListening(speechIntent);
+                tts.Stop();
+                tts.Shutdown();
             }
+            base.OnStop();
+        }
+
+        private bool IsTalkBackEnabled()
+        {
+            var accessManager = (AccessibilityManager)GetSystemService(AccessibilityService);
+            return accessManager.IsTouchExplorationEnabled;
```

Comment:
If TB is enabled, we dont want to be voicing or doing any voice commands, because TB will f*** everything up big time. And why would we? With TB blind people can use our app with no problems.